lib/Util.ts

```diff
@@ -107,17 +107,17 @@ export async function decompressGzippedContent(buffer: Buffer, charset?: string)
  * @return {RegExp}
  */
 export function buildProxyBypassRegexFromEnv(bypass : string) : RegExp {
-    try {
-        // We need to keep this around for back-compat purposes
-        return new RegExp(bypass, 'i')    
-    }
-    catch(err) {
-        if (err instanceof SyntaxError && (bypass || "").startsWith("*")) {            
-            let wildcardEscaped = bypass.replace('*', '(.*)');
-            return new RegExp(wildcardEscaped, 'i');
-        }
-        throw err;
+    const searchRegExpToReplaceSpecialChars: RegExp = new RegExp('([.])', 'g');
+
+    // replace all . symbols in string by \. because point is a special character
+    let safeRegex: string = (bypass || "").replace(searchRegExpToReplaceSpecialChars, '\\$1');
```

Comment:
Won't it break customers which use regexp as a value with dots as special characters?

Reply:
Generally NO_PROXY env variable shouldn't contain a regex inside, the strings the set inside should be interpreted as is. The reason why we need to escape dot symbols is that we look for the specific domains in URLs using the RegExp test function which accepts a regular expression as a pattern to search, so we transform the string to the RegExp. NO_PROXY is quite a common way of setup a proxy bypass list and as far as I see there is no soft that supports regexp inside, so dots shouldn't be used as special characters there and in my mind, they could be safely escaped.

Reply:
And also our docs related to the setup of proxy bypass domains is saying that dots should be escaped - https://docs.microsoft.com/en-us/azure/devops/pipelines/agents/proxy?view=azure-devops&tabs=windows#specify-proxy-bypass-urls